src/encoder/stateless/av1/vaapi.rs

```diff
@@ -0,0 +1,975 @@
+            CodedOutputPromise::new(picture, references, coded_buf, request.coded_output);
+
+        Ok((reference_promise, bitstream_promise))
+    }
+}
+
+impl<M, H> StatelessEncoder<AV1, H, VaapiBackend<M, H>>
+where
+    M: SurfaceMemoryDescriptor,
+    H: std::borrow::Borrow<libva::Surface<M>> + 'static,
+{
+    pub fn new_vaapi(
+        display: Rc<libva::Display>,
+        config: EncoderConfig,
+        fourcc: Fourcc,
+        coded_size: Resolution,
+        low_power: bool,
+        blocking_mode: BlockingMode,
+    ) -> EncodeResult<Self> {
+        let va_profile = match config.profile {
+            Profile::Profile0 => VAProfileAV1Profile0,
+            Profile::Profile1 => VAProfileAV1Profile1,
+            _ => return Err(StatelessBackendError::UnsupportedProfile.into()),
+        };
+
+        let backend = VaapiBackend::new(
+            display,
+            va_profile,
+            fourcc,
+            coded_size,
+            libva::constants::VA_RC_CQP,
+            low_power,
+        )?;
+
+        Self::new_av1(backend, config, blocking_mode)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use libva::constants::VA_RT_FORMAT_YUV420;
+    use libva::Display;
+    use libva::UsageHint;
+    use libva::VAEntrypoint::VAEntrypointEncSliceLP;
+    use libva::VAProfile::VAProfileAV1Profile0;
+
+    use super::*;
+    use crate::backend::vaapi::encoder::tests::upload_test_frame;
+    use crate::backend::vaapi::encoder::tests::TestFrameGenerator;
+    use crate::backend::vaapi::surface_pool::PooledVaSurface;
+    use crate::backend::vaapi::surface_pool::VaSurfacePool;
+    use crate::codec::av1::parser::CdefParams;
+    use crate::codec::av1::parser::ColorConfig;
+    use crate::codec::av1::parser::FrameHeaderObu;
+    use crate::codec::av1::parser::FrameType;
+    use crate::codec::av1::parser::ObuHeader;
+    use crate::codec::av1::parser::ObuType;
+    use crate::codec::av1::parser::OperatingPoint;
+    use crate::codec::av1::parser::QuantizationParams;
+    use crate::codec::av1::parser::SequenceHeaderObu;
+    use crate::codec::av1::parser::TemporalDelimiterObu;
+    use crate::codec::av1::parser::TileInfo;
+    use crate::codec::av1::parser::TxMode;
+    use crate::codec::av1::parser::MAX_NUM_OPERATING_POINTS;
+    use crate::codec::av1::parser::PRIMARY_REF_NONE;
+    use crate::codec::av1::parser::SELECT_INTEGER_MV;
+    use crate::codec::av1::parser::SUPERRES_NUM;
+    use crate::codec::av1::synthesizer::Synthesizer;
+    use crate::decoder::FramePool;
+    use crate::encoder::stateless::simple_encode_loop;
+    use crate::encoder::stateless::BackendPromise;
+    use crate::encoder::stateless::StatelessEncoderBackendImport;
+    use crate::encoder::FrameMetadata;
+    use crate::utils::IvfFileHeader;
+    use crate::utils::IvfFrameHeader;
+    use crate::FrameLayout;
+    use crate::PlaneLayout;
+    use crate::Resolution;
+
+    #[test]
+    // Ignore this test by default as it requires libva-compatible hardware.
```

Comment:
`#[ignore]` was missing here, added it as otherwise this test runs by default and would fail on the Github runners.